contracts/contracts/strategies/NativeStaking/CompoundingValidatorManager.sol

```diff
@@ -0,0 +1,865 @@
+    ) external onlyRegistrator {
+        bytes32 consolidationLastPubKeyHashMem = consolidationLastPubKeyHash;
+        require(
+            consolidationLastPubKeyHashMem != bytes32(0),
+            "No consolidations"
+        );
+
+        bytes32 blockRoot = BeaconRoots.parentBlockRoot(parentBlockTimestamp);
+        // Verify the validator index has the same public key as the last source validator
+        IBeaconProofs(BEACON_PROOFS).verifyValidatorPubkey(
+            blockRoot,
+            consolidationLastPubKeyHashMem,
+            validatorPubKeyProof,
+            lastValidatorIndex
+        );
+
+        // Verify the balance of the last validator in the consolidation batch
+        // is zero. If its not then the consolidation has not been completed.
+        // This proof is to the beacon block root, not the balances container root.
+        uint256 validatorBalance = IBeaconProofs(BEACON_PROOFS)
+            .verifyValidatorBalance(
+                blockRoot,
+                balancesLeaf,
+                validatorBalanceProof,
+                lastValidatorIndex,
+                IBeaconProofs.BalanceProofLevel.BeaconBlock
+            );
+        require(validatorBalance == 0, "Last validator balance not zero");
+
+        // Call the old sweeping strategy to confirm the consolidation has been completed.
+        // This will decrease the balance of the source strategy by 32 ETH for each validator being consolidated.
+        uint256 consolidationCount = IConsolidationSource(
+            consolidationSourceStrategy
+        ).confirmConsolidation();
+
+        // Increase the balance of this strategy by 32 ETH for each validator being consolidated.
+        // This nets out the decrease in the source strategy's balance.
+        lastVerifiedEthBalance += SafeCast.toUint128(
+            consolidationCount * 32 ether
+        );
+
+        // Reset the stored consolidation state
+        consolidationLastPubKeyHash = bytes32(0);
+        consolidationSourceStrategy = address(0);
+
+        emit ConsolidationVerified(
+            consolidationLastPubKeyHashMem,
+            lastValidatorIndex,
+            consolidationCount
+        );
+
+        // Unpause now the balance of the target validator has been verified
+        _unpause();
+
+        // Take a snap of the balances so the actual balances of the new validator balances can be verified
+        _snapBalances();
+    }
+
+    /// @notice Stores the current ETH balance at the current block.
+    /// The validator balances on the beacon chain can then be proved with `verifyBalances`.
+    /// Can not be called while a consolidation is in progress.
+    function snapBalances() public whenNotPaused {
+        _snapBalances();
+    }
+
+    function _snapBalances() internal {
+        // Check no consolidation is waiting to be verified
+        require(
+            consolidationLastPubKeyHash == bytes32(0),
+            "Consolidation in progress"
+        );
```

Comment:
I think this require is not reachable (i.e. it will never revert).

We are looking for a situation where `consolidationLastPubKeyHash != bytes32(0)`. 
This can happen only in the situation where `requestConsolidation()` has been called. But when `requestConsolidation()` is called, it `pause` the contract which prevent the `snapBalance` to be called.

But still not 100% sure. What do you think?